test/horreum_client_it.py

```diff
@@ -0,0 +1,101 @@
+
+from horreum.horreum_client import new_horreum_client, HorreumClient
+from horreum.raw_client.api.test.test_request_builder import TestRequestBuilder
+from horreum.raw_client.models.protected_type_access import ProtectedType_access
+from horreum.raw_client.models.test import Test
+
+username = "user"
+password = "secret"
+
+
+@pytest.fixture()
+async def anonymous_client_without_check() -> HorreumClient:
+    print("Setting up anonymous client")
+    return await new_horreum_client(base_url="http://localhost:8080")
+
+
+@pytest.fixture()
+async def anonymous_client() -> HorreumClient:
+    print("Setting up anonymous client")
+    client = await new_horreum_client(base_url="http://localhost:8080")
+    try:
+        await client.raw_client.api.config.version.get()
+    except httpx.ConnectError:
+        pytest.fail("Unable to fetch Horreum version, is Horreum running in the background?")
+    return client
+
+
+@pytest.fixture()
+async def authenticated_client() -> HorreumClient:
+    print("Setting up authenticated client")
+    client = await new_horreum_client(base_url="http://localhost:8080", username=username, password=password)
+    try:
+        await client.raw_client.api.config.version.get()
+    except httpx.ConnectError:
+        pytest.fail("Unable to fetch Horreum version, is Horreum running in the background?")
+    return client
+
+
+def test_check_client_version(anonymous_client_without_check: HorreumClient):
+    version = anonymous_client_without_check.version()
+    # TODO: we could load the toml and check the versions match
+    assert version != ""
+
+
+@pytest.mark.asyncio
+async def test_check_server_version(anonymous_client: HorreumClient):
+    version = await anonymous_client.raw_client.api.config.version.get()
+    assert version.version != ""
+    assert version.start_timestamp != ""
+
+
+@pytest.mark.asyncio
+async def test_check_missing_token(anonymous_client: HorreumClient):
+    req = RequestInformation(Method("GET"), "/api/")
+    await anonymous_client.auth_provider.authenticate_request(req)
+    assert len(req.headers.get(BaseBearerTokenAuthenticationProvider.AUTHORIZATION_HEADER)) == 0
+
+
+@pytest.mark.asyncio
+async def test_check_auth_token(authenticated_client: HorreumClient):
+    req = RequestInformation(Method("GET"), "/api/")
+    await authenticated_client.auth_provider.authenticate_request(req)
+    assert len(req.headers.get(BaseBearerTokenAuthenticationProvider.AUTHORIZATION_HEADER)) == 1
+    assert req.headers.get(BaseBearerTokenAuthenticationProvider.AUTHORIZATION_HEADER).pop().startswith("Bearer")
+
+
+@pytest.mark.asyncio
+async def test_missing_username_with_password():
+    try:
+        await new_horreum_client(base_url="http://localhost:8080", password=password)
+        pytest.fail("expect RuntimeError here")
+    except RuntimeError as e:
+        assert str(e) == "providing password without username, have you missed something?"
```

Comment:
As [I mentioned](https://github.com/Hyperfoil/horreum-client-python/pull/2#discussion_r1534639555), there's PyTest support for this.

Reply:
Sorry, I forgot about that.
Updated now